### DISCUSSION TITLE
How to install Magic-link Extension in Keycloak with Kubernetes installation

### DISCUSSION THREAD

**Discussion by Adarsh505-cloud** [keycloak/keycloak#19856] [Q&A]
Hi 

I have installed the Keycloak using the bitnami/keycloak chart and it seems working fine, but I couldn’t find an option to install additional Extensions like keycloak-magic-link. We need to install an add-on.

I could see a git to build the jar file, but not sure how can we use it inside the Kubernetes pods, can you please add the instruction to install Extensions also. I can’t find an example for the same, no documentation is available on the public internet to solve this issue.

Extensions link from the official Keycloak site: https://github.com/p2-inc/keycloak-magic-link

Where can I get the file show we build it using mvm clean install command as instructed in the doc? if so we have the jar file on how to build it inside the Keycloak container.

I couldn’t see a standalone/deployment directory inside the Keycloak directory. Please help me if any one familiar with the setup.

Thanks

**Comment by darius-m** [comment]
I think the `standalone/deployment` is a reference to the deployment method for older Keycloak versions, that were built on top of WildFly (I think it was the default up until Keycloak 18). For current versions, based on Quarkus, you add the jar files to the `providers/` directory and then run the `kc.sh` script with the `build` option, as mentioned in the server developer guide - https://www.keycloak.org/docs/latest/server_development/index.html#packaging-classes-and-deployment-2.

**Comment by Adarsh505-cloud** [comment]
Hi,

I have alredy done the given task manually, directly copied the mentioned
jar file inside the kubernets pod and run the kc.sh build command, but I
could not see the magic link extension in the console, thats why I asked
for instructions to install.
